File: backend/doc_configs/sla.py
from typing import Optional
from pydantic import BaseModel
from doc_configs.shared import PartialPartyInfo


SYSTEM_PROMPT = """You are a legal document assistant helping a user fill in a Service Level Agreement (SLA) Cover Page.

## Fields you must collect

1. **Target Uptime** -- minimum uptime percentage for the cloud service
   Ask: "What target uptime should the provider commit to? (e.g. '99.9%', '99.95%', '99.99%')"

2. **Uptime Credit** -- service credit if uptime falls below target
   Ask: "What service credit should apply when uptime falls below target? (e.g. '5% of monthly fees per 1% below target', '10% credit for each hour of downtime')"

3. **Measurement Period** -- how uptime and response times are measured
   Ask: "What measurement period should be used for calculating uptime? (e.g. 'Calendar month', 'Calendar quarter', 'Rolling 30-day period')"

4. **Party 1 (Provider)** -- name, title, company, notice address, signing date
   Ask for all sub-fields. Example: "Jane Smith, CTO of CloudCo, jane@cloudco.com, signing today"

5. **Party 2 (Customer)** -- same sub-fields as Party 1

For date shortcuts like "today", "tomorrow", "Mon", "next Mon", use this lookup table:
{date_lookup}
For full dates, convert to YYYY-MM-DD format directly.

## Current document state
{current_fields}

## Instructions
- Work through the fields in order, but adapt naturally to what the user already told you
- Ask 1-2 questions per turn maximum
- Always include a concrete example in parentheses when asking a question
- When the user answers, confirm what you captured then always end with a follow-up question
- Once all fields are collected, tell the user the document is complete and they can download the PDF

## Clearing fields
If the user asks to clear fields (e.g. "clear all", "reset", "clear the uptime"):
- To clear a specific field, return that field set to its empty value and confirm what was cleared
- To clear all fields, return every field reset to empty values, confirm, then ask the first question again
- Never clear silently

Return JSON with:
- "message": your conversational reply (include examples when asking questions)
- "fields": only the fields you are confident about from this turn; omit everything else
"""


class PartialFields(BaseModel):
    targetUptime: Optional[str] = None
    uptimeCredit: Optional[str] = None
    measurementPeriod: Optional[str] = None
    party1: Optional[PartialPartyInfo] = None
    party2: Optional[PartialPartyInfo] = None


class AiResponse(BaseModel):
    message: str
    fields: PartialFields

File: frontend/lib/schemas/sla.ts
import { DocSchema } from "../doc-schema";

export const slaSchema: DocSchema = {
  id: "sla",
  title: "Service Level Agreement",
  subtitle: "Cover Page",
  infoBox:
    'This Service Level Agreement (the "SLA") consists of: (1) this Cover Page and (2) the Common Paper SLA Standard Terms Version 2.0 identical to those posted at commonpaper.com/standards/service-level-agreement/2.0. The SLA is designed to be used alongside a Cloud Service Agreement to define uptime targets, response times, and remedies.',
  coverFields: [
    {
      key: "targetUptime",
      label: "Target Uptime",
      hint: "The minimum uptime percentage for the cloud service",
      type: "text",
      placeholder: "e.g. 99.9%",
    },
    {
      key: "uptimeCredit",
      label: "Uptime Credit",
      hint: "Service credit if uptime falls below target",
      type: "text",
      placeholder: "e.g. 5% of monthly fees per 1% below target",
    },
    {
      key: "measurementPeriod",
      label: "Measurement Period",
      hint: "How uptime and response times are measured",
      type: "text",
      placeholder: "e.g. Calendar month",
    },
  ],
  parties: [
    { label: "Provider", fields: ["name", "title", "company", "noticeAddress", "date"] },
    { label: "Customer", fields: ["name", "title", "company", "noticeAddress", "date"] },
  ],
  signingNote: "By signing this Cover Page, each party agrees to enter into this SLA as of the Effective Date.",
  templateFile: "templates/sla.md",
  footerNote: "Common Paper Service Level Agreement (Version 2.0) free to use under CC BY 4.0.",
  greeting:
    "Hi! I'm your SLA assistant. I'll help you fill in the Service Level Agreement Cover Page through conversation. This defines uptime commitments, response times, and service credits for a cloud service. Let's start -- what target uptime percentage should the provider commit to? (e.g. '99.9%', '99.95%', '99.99%')",
  pdfFilename: "service-level-agreement.pdf",
};
